src/server/server.test.js

```diff
@@ -0,0 +1,54 @@
+import request from 'supertest';
+import * as after from '@jaredpalmer/after';
+import server from './index';
+
+describe('Server', () => {
+  const makeRequest = async path => request(server).get(path);
+
+  it(`should not pass an 'x-powered-by' response header`, async () => {
+    const { headers } = await makeRequest('/status');
+    const headerKeys = Object.keys(headers);
+    expect(headerKeys).not.toContain('x-powered-by');
+  });
+
+  describe('/status', () => {
+    it('should respond with a 200', async () => {
+      const { statusCode } = await makeRequest('/status');
+      expect(statusCode).toBe(200);
+    });
+  });
+
+  describe('/*', () => {
+    const testResponseText = (testTitle, value, mockRender, testAssertion) => {
+      it(testTitle, async () => {
+        after.render = jest.fn().mockImplementationOnce(mockRender);
+        const { text } = await makeRequest('/');
+        testAssertion(text, value);
+      });
+    };
+
+    describe('Successful render', () => {
+      const testValue = 'data';
+      testResponseText(
+        'should respond with rendered data',
+        testValue,
+        () => testValue,
+        (text, value) => expect(text).toContain(value),
+      );
```

Comment:
I have a suggestion that is about the readability of these tests than the functionality.  
By abstracting tests out into the `testResponseText` function, it makes it more cognitive overhead to parse through the order the test is going through.
I would suggest restructuring things a bit to make this more easily readable.

For example, instead of pulling out all the title and assertions out, just having a render function that handles this.

```
    const renderWithTestData = mockRender => {
      after.render = jest.fn().mockImplementationOnce(mockRender);
    };

    describe('Successful render', () => {
      it('should respond with rendered data', async () => {
        const testData = 'data';

        renderWithTestData(() => testData);

        const { text } = await makeRequest('/');
        expect(text).toContain(testData);
      });
    });
```

Reply:
I wrote them in such a way to avoid CC complaining about duplication. Your way looks cleaner, will try it